### PR TITLE
Implement redux form

### PR DESCRIPTION
**Pull request by ivan-unfolds**

### CODE REVIEW

**Comment by jay-meister** [comment]
Hey @rjmk would it be possible to exclude the membership_type object from the MembersController accountInfo function?  Ideally we would have membership_type _String_ which comes from the members table but i think it is getting overwritten.

**Comment by rjmk** [comment]
@JMurphyWeb It is certainly possible. It's probably there because we're likely to need the amount of the subscription payment. You can get the `value` off the object if you want the name of the type.

Alternatively, you could change `populateAll()` on line 22 to `populate('payments')`

**Comment by ivan-unfolds** [comment]
@rjmk this pull request is not ready to be merged in, however if you have some time could you have a look, see if we are going in right direction? #299 has what we are working on atm.

File: node_modules/app/http.js [hunk 1]
```diff
@@ -2,16 +2,22 @@
 
 var Task = require('data.task')
 var xhr = require('xhr')
-var { curry, prop, compose, map } = require('ramda')
+var { curry, prop, compose, map, identity } = require('ramda')
 
-var get = (path) => request('get', path, {})
-
-var post = curry((body, path) => request('post', path, { json: body }))
+var basic_request = curry((method, body, path) =>
```
Comment (rjmk) [inline node_modules/app/http.js:7]: Let's not do this, as it leaves us with multiple APIs for async actions, which is confusing

File: src/shared/form_fields/user_my_details.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
+
+export const contact_fields =
```
Comment (rjmk) [inline src/shared/form_fields/user_my_details.js:2]: I think we should have moved `src/admin/form_fields/member.js` here and built on that

File: src/user/components/my_details/details_form.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import React from 'react'
```
Comment (rjmk) [inline src/user/components/my_details/details_form.js:1]: I'm surprised how little this file and, to a lesser extent, the reducer make use of the member view from `admin`. It seems like that would save you a lot of work

**Pull request closed without merging**